### PR TITLE
chore: contract now sells item at auction on user's behalf

### PR DESCRIPTION
**Pull request by katelynsills**
This PR changes the contract to sell the item at auction on the user's behalf rather than return the item and request that the user sell the item at auction, to get around https://github.com/Agoric/agoric-sdk/issues/2043. The user's proposal changes from:
```js
    const AliceListingOffer = harden({
      want: { Items: show2 },
      give: { Money: videoService.pricePerItem() },
});
```
to:
```js
    const AliceListingOffer = harden({
      want: { AuctionProceeds: makeMoola(reservePrice) },
      give: { ListingFee: videoService.pricePerItem() },
    });
```
And the `makeListingInvitation` call changes to:

```js
    const buyListingInvitation = videoService.makeListingInvitation(
      show2,
      timeAuthority,
      closesAfter,
    );
```

It also allows the creator to withdraw listing fees at any time by making an invitation (`E(creatorFacet).makeWithdrawFeesInvitation()`). The creator can be notified if there are any new fees to be withdrawn by getting a notifier (`E(creatorFacet).getFeesAccumulatedNotifier()`, and calling `E(notifier).getUpdateSince()`

@ski and @dckc let me know if you have any questions. I'd be happy to walk through the changes and help hook this up to the dapp handler and UI.

### CODE REVIEW

**Comment by dckc** [comment]
wow! very nice. I think I/we wished for a design like this earlier but didn't see how to do it.

File: contract/src/contract.js [hunk 1]
```diff
@@ -15,104 +21,184 @@ import { E } from '@agoric/eventual-send';
 const start = async (zcf) => {
   // Terms are
   const {
-    issuers: { Money: moolaIssuer },
+    issuers: { AuctionProceeds: auctionProceedsIssuer },
+    maths: { ListingFee: listingFeeMath },
```
Comment (dckc) [inline contract/src/contract.js:25]: splitting the money issuer into separate issuers for listing fees and auction proceeds is more than we need, but I guess it doesn't hurt much.

If I were going to split the auction proceeds issuer, I'd want to do it on a per-auction basis.

**Review by dckc — APPROVED** [review]

File: contract/src/contract.js (hunk 1 above)
Comment (katelynsills) [reply contract/src/contract.js:25]: > splitting the money issuer into separate issuers for listing fees and auction proceeds is more than we need, but I guess it doesn't hurt much.

Yeah, Zoe doesn't allow the same keyword to be used under both `want` and `give`, so it's required if we want Alice to  have offer safety over giving money and receiving money from the auction.

Comment (dckc) [reply contract/src/contract.js:25]: ah. so it's *not* more than we need.

**Comment by dckc** [comment]
@katelynsills can Alice cancel her offer after bob starts the auction but before the auction closes? If so, then Bob is out the cost of running the auction but doesn't collect a listing fee.

Is this what `exit: { waived: null }` is about? We need that in `AliceListingOffer` then too, right?

**Comment by katelynsills** [comment]
> @katelynsills can Alice cancel her offer after bob starts the auction but before the auction closes? If so, then Bob is out the cost of running the auction but doesn't collect a listing fee.

> Is this what `exit: { waived: null }` is about? We need that in `AliceListingOffer` then too, right?

Yeah, you could require all listing offers to have `exit: { waived: null }` and then Alice wouldn't be able to cancel her offer before the auction ended. That makes Alice vulnerable to her money getting stuck in the contract (waives payout liveness), so that's a downside.

**Comment by dckc** [comment]
By design Alice should have no expectation of getting the listing fee back once her show is listed. So it's not much of a downside. In fact, Bob (the house) should get the listing fee even if nobody bids. So we might go with this PR for now but it's not quite right either.

**Comment by katelynsills** [comment]
> By design Alice should have no expectation of getting the listing fee back once her show is listed. So it's not much of a downside. In fact, Bob (the house) should get the listing fee even if nobody bids. So we might go with this PR for now but it's not quite right either.

Sounds good! Feel free to take as much as is helpful.

**Pull request merged**